### PR TITLE
Hotfix/update component moderation state [ENG-2804]

### PR DESCRIPTION
**Pull request by jwalz**
## Purpose
Fix bugs where nested registrations don't get their `moderation_state` updated and where registrations weren't getting the correct `moderation_state` after successfully completing an Embargo.
## Changes
* Modified `update_moderation_state` to set the `moderation_state` for all nodes in the registration tree
* Modified `terminate_embargo` to refresh from the DB after invoking the state machine to complete the Embargo
  * State machine ultimately updates state on embargo.target_registration, which isn't recognized locally. Further local changes were overwriting the new value.
* Tests to confirm the correct state changes in both of these cases
* Changes to RegistrationFactory to make it easier to create Registration trees.  

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify
- Verify

What are the areas of risk?

Any concerns/considerations/questions that development raised?

### CODE REVIEW

File: osf_tests/factories.py [hunk 1]
```diff
@@ -411,14 +411,15 @@ def _create(cls, target_class, project=None, is_public=False,
...
-                provider=provider
+                provider=provider,
             )
         auth = Auth(user=user)
         register = lambda: project.register_node(
             schema=schema,
             auth=auth,
             draft_registration=draft_registration,
             provider=provider,
+            parent=kwargs.get('parent')
```
Comment (corbinSanders) [inline osf_tests/factories.py:422]: Should this be kwargs.get('parent', None')`?

**Review by corbinSanders — CHANGES_REQUESTED** [review]
One comment. Otherwise LGTM!

File: osf_tests/factories.py (hunk 1 above)
Comment (jwalz) [reply osf_tests/factories.py:422]: `get` intercepts Key errors and returns None by default. I'm cool being more explicit if that's the norm elsewhere

Comment (corbinSanders) [reply osf_tests/factories.py:422]: I'm good with that. Thanks!

**Pull request merged**